.eslintrc.json

```diff
@@ -1,8 +1,48 @@
 {
-  "plugins": ["import"],
-  "extends": ["react-app", "react-app/jest", "prettier"],
+  "env": {
+    "browser": true,
+    "es2021": true
+  },
+  "settings": {
+    "react": {
+      "version": "detect"
+    }
+  },
+  "extends": [
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:@typescript-eslint/recommended",
+    "prettier"
+  ],
   "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "ecmaVersion": "latest",
+    "sourceType": "module"
+  },
+  "plugins": ["react", "@typescript-eslint", "import", "react-hooks"],
+  "overrides": [
+    {
+      "files": ["config-overrides.js", "jest.config.js"],
+      "env": {
+        "node": true,
+        "commonjs": true
+      }
+    }
+  ],
   "rules": {
+    "@typescript-eslint/no-var-requires": "off",
```

Comment:
Ty, for commenting and giving me some direction. This was the initial stack these changes were included. I should have been a little more direct about asking for feedback. https://github.com/jbx-protocol/juice-interface/pull/876#issue-1216741357

I wasn't sure what the team wanted, and moving forward didn't want to change the code base too much, so I figured to do what was most sensible without introducing too many new rules right off the bat. I agree with most of your comments. I just wanted to get the ✅ before making all of these changes.

> One comment to cover some rules here:
> 
> * `@typescript-eslint/no-var-requires` - Do we want to allow requires? Isn't it better to use ES2015-style imports and use a disable if needed?

Yes, let's turn this one on.


> * `@typescript-eslint/ban-type` - I think we want to keep the [best practices](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-types.md#default-options) here, wdyt?

👍🏻 -- here for it.

> * `react/no-children-prop` - i am not as well versed in react as others, but it is sometimes confusing and easier to miss children when passed in as props, I think it would be better to leverage the JSX and pass them in as DOM children.
👍🏻 SAME


> * `react/no-unescaped-entities` - This I think is a good one to keep to avoid accidentally leaking details into code.
👍🏻

> * `react/prop-types` - I would prefer to keep this one. I think it is less an issue with typescript as it likely already enforces it.
👍🏻

> * `prefer-const` - This one I think we should keep on too. Is there any benefit to not preferring const?
👍🏻

> * `jsx-a11y/img-redundant-alt` - seems like this is a good one to keep on for accessibility enforcement
👍🏻
> * `react/react-in-jsx-scope` - we definitely want this one I think to ensure that react is in scope

This one I don't think we need anymore: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md#when-not-to-use-it it would be dope to remove some boilerplate. But I'm not married to the idea either.

Reply:
Oh awesome! I didn't actually realise that these violations already existed in the code. I am more leaning toward just pulling the trigger and fixing them in the PR - wdyt @aeolianeth ?

Reply:
@benschac @wraeth-eth i think we should resolve them but I don't want to block this PR any longer- happy for these changes to come in a follow-up